backend/app/Http/Controllers/ResultController.php

```diff
@@ -23,9 +31,29 @@ public function index()
      * @param  \Illuminate\Http\Request  $request
      * @return \Illuminate\Http\Response
      */
-    public function store(Request $request)
+    public function store(User $user, Category $category, Request $request)
     {
-        //
+        $data = $request->validate([
+            'user_progress_id' => ['required', 'integer', 'exists:user_progress,id'],
+            'questions' => ['required', 'array'],
+            'questions.*' => ['required', 'integer', 'exists:questions,id'],
+            'answers' => ['required', 'array'],
+            'answers.*' => ['required', 'integer', 'exists:choices,id'],
+            'results' => ['required', 'array'],
+            'results.*' => ['required', 'boolean']
+        ]);
+
+        $results = [];
+        for ($i = 0; $i < count($request->get('questions')); $i++) {
+            $result['user_progress_id'] = $data['user_progress_id'];
+            $result['category_id'] = $category->id;
+            $result['question_id'] = $data['questions'][$i];
+            $result['user_choice_id'] = $data['answers'][$i];
+            $result['is_correct'] = $data['results'][$i];
+            $results[] = Result::create($result);
+        }
```

Comment:
Suggestion: You can map the result using the `map()` like the JS, this is related to [PHP Collections](https://laravel.com/docs/8.x/collections)